Adding and removing the SRIOV ROCE VF port

PavamanSubramaniyam opened the pull request:
Adding the code to support the adding and removal of single sriov roce vf port

PavamanSubramaniyam commented:
Execution results for the executed tests is pasted here. Detailed logs are attached in the Jira task.

# cat sriov_device_test.py.data/sriov_device_test.yaml
hmc_username: "hscroot"
hmc_pwd: ""
sriov_adapter: "U78D8.ND0.FGD0047-P0-C4-C0"
sriov_port: "0"
sriov_roce: "1"
ipaddr: "17.1.1.103"
netmasks: "255.255.255.0"
peer_ip: "17.1.1.183"
mac_id: "ba:70:ce:95:cc:03"

# avocado run --max-parallel-tasks=1 sriov_device_test.py -m sriov_device_test.py.data/sriov_device_test.yaml
JOB ID     : d2c08c1b68a96023118195b318c4642f8e7be817
JOB LOG    : /root/avocado-fvt-wrapper/results/job-2023-09-06T23.42-d2c08c1/job.log
 (1/4) sriov_device_test.py:NetworkSriovDevice.test_add_logical_device;run-5f18: STARTED
 (1/4) sriov_device_test.py:NetworkSriovDevice.test_add_logical_device;run-5f18: PASS (11.66 s)
 (2/4) sriov_device_test.py:NetworkSriovDevice.test_add_migratable_sriov;run-5f18: STARTED
 (2/4) sriov_device_test.py:NetworkSriovDevice.test_add_migratable_sriov;run-5f18: CANCEL: Test unsupported (0.84 s)
 (3/4) sriov_device_test.py:NetworkSriovDevice.test_remove_migratable_sriov;run-5f18: STARTED
 (3/4) sriov_device_test.py:NetworkSriovDevice.test_remove_migratable_sriov;run-5f18: CANCEL: Test unsupported (0.84 s)
 (4/4) sriov_device_test.py:NetworkSriovDevice.test_remove_logical_device;run-5f18: STARTED
 (4/4) sriov_device_test.py:NetworkSriovDevice.test_remove_logical_device;run-5f18: PASS (13.94 s)
RESULTS    : PASS 2 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 2
JOB HTML   : /root/avocado-fvt-wrapper/results/job-2023-09-06T23.42-d2c08c1/results.html
JOB TIME   : 49.05 s

abdhaleegit commented:
@PavamanSubramaniyam looks like travis fails for indent issues,, check line 161 and later for proper spaces

PavamanSubramaniyam commented:
Fixed the Travis fail for indent issues.

abdhaleegit commented on io/net/sriov_device_test.py:
```diff
@@ -70,6 +70,8 @@ def setUp(self):
                                              '*', default=None).split(' ')
         self.sriov_port = self.params.get('sriov_port', '*',
                                           default=None).split(' ')
+        self.sriov_roce = self.params.get('sriov_roce', '*',
```
Please remove the path param  '*'

abdhaleegit replied:
@PavamanSubramaniyam remove it for all the parameters

abdhaleegit reviewed (changes requested):
@PavamanSubramaniyam Please create separate yaml for sriov_roce  device test and also add in readme about new parameter usage with sample values

PavamanSubramaniyam commented:
I have created separate yaml for sriov_roce device test and also added in the readme about new parameter usage with the sample value.  Also, removed the path param '*' for all the parameters being passed.

abdhaleegit commented on io/net/sriov_device_test.py.data/README.txt:
```diff
@@ -6,6 +6,7 @@ hmc_username = Specify the HMC user name.
 hmc_pwd = Specify the HMC password.
 sriov_adapter = Specify the sriov adapter loc code for test.
 sriov_port: specify adapter port for test.
+sriov_roce: specify if the roce adapter VF needs to be added and removed ex: "1"
```
@PavamanSubramaniyam Please make sure it is boolean like True or False instead numeric

PavamanSubramaniyam replied:
I have made the changes as suggested.

abdhaleegit replied:
I tested this, but fails.. as zip function does not iterate boolean values.. so better to have string or numeric

abdhaleegit commented on io/net/sriov_device_test.py:
```diff
@@ -67,13 +67,15 @@ def setUp(self):
-        self.ipaddr = self.params.get('ipaddr', '*', default="").split(' ')
-        self.netmask = self.params.get('netmasks', '*', default="").split(' ')
+        self.sriov_roce = self.params.get('sriov_roce',
+                                          default=None).split(' ')
```
if sriov_roce param is not given from other yamls.. the test case will fail.. so please add a check to see if param in yaml

abdhaleegit replied:
no need to split boolean valu, also better to check if param is given else other yamls will fail if it does not have parameter

```
        if 'sriov_roce' in self.params:
            self.sriov_roce = self.params.get('sriov_roce', default=None)
```

PavamanSubramaniyam replied:
I have incorporated the review comments.

abdhaleegit replied:
ignore my previous comment, the params are not iterable so it wont work

abdhaleegit commented:
@PavamanSubramaniyam will accept the original changes as is

abdhaleegit commented:
@PavamanSubramaniyam Please test
1. by removing the roce parameter passing in all function and instead making it class parameter flag as self and using it everywhere
2. if above is not working.. upload logs for rebased code

Naresh-ibm commented on io/net/sriov_device_test.py:
```diff
@@ -249,13 +253,13 @@ def test_remove_logical_device(self):
-        for mac, slot in zip(self.mac_id, self.sriov_adapter):
-            logical_port_id = self.get_logical_port_id(mac)
+        for mac, slot, sriov_roce in zip(self.mac_id, self.sriov_adapter, self.sriov_roce):
+            logical_port_id = self.get_logical_port_id(mac, sriov_roce)
```
what if the `sriov_roc` parameter is not there in yaml?
what value `self.sriov_roce` will hold here?

Naresh-ibm commented on io/net/sriov_device_test.py.data/README.txt:
```diff
@@ -7,6 +7,7 @@ hmc_username = Specify the HMC user name.
 hmc_pwd = Specify the HMC password.
 sriov_adapter = Specify the sriov adapter loc code for test.
 sriov_port: specify adapter port for test.
+sriov_roce: specify if the roce adapter VF port needs to be added and removed ex: True
```
sriov_roce: specify  `True` if the roce adapter VF port needs to be added and removed else `False`

Naresh-ibm replied:
@just change this rest looks good

Naresh-ibm reviewed (approved):
LGTM

abdhaleegit reviewed (approved):
LGTM
